addons/mail/static/src/core/common/chat_window_service.js

```diff
@@ -125,7 +125,7 @@ export class ChatWindowService {
     makeVisible(chatWindow) {
         const swaped = this.visible[this.visible.length - 1];
         this.hide(swaped);
-        this.show(chatWindow);
+        this.show(chatWindow, { notifyState: false });
```

Comment:
Shouldn't this still notify if the previous state was folded, as it will effectively change it to open?

Reply:
If we do that, this will swap the chat window on the other tabs. I think we should either synchronize hidden chat windows properly or not do it at all: how to explain that making a chat window visible sometimes swap it on other tabs and sometimes not? It will probably look like a bug. That's why I disabled it in the `makeVisible` case.

Reply:
Isn't that a bug though?
Changing fold state shouldn't lead to doing swap?

Reply:
Or maybe we should sent on the bus something else that is not just open but make visible?

Reply:
Right now we don't have any way to tell whether an  `open` event results from unfolding or making a chat window visible. I think syncing the  hidden state is an improvement not a fix since it was already behaving this way in 16.0. Now we have to decide whether we wan't to keep partial support or not supporting it at all.